application/services/Reports.php

```diff
@@ -6532,4 +6636,131 @@ public function getXtptIndicatorsReport($params) {
+                                }
+                                if ($noResultOrError2127InSubmission) {
+                                    array_push($allComments, 'Check computer software GeneXpert Dx to ensure software is not freezing during testing. Check uninterrupted power supply (UPS) to ensure it is capable of sustaining power to GeneXpert for a minimum of 2 hours should an electrical power outage occur. If UPS is unable to sustain GeneXpert instrument for 2 hours request sufficient UPS that will provide power to GeneXpert instrument for no less than 2 hours to enable current run completion in the event of a power outage.');
+                                }
+                                if ($result['instrument_requires_calibration']) {
+                                    array_push($allComments, 'Calibration is an important maintenance procedure to ensure GeneXpert instruments are functioning properly and yielding accurate results. '.$result['tests_done_after_calibration_due'].'Instrument calibration should take place each year or after every 2,000 runs on each instrument module (whichever comes first). Entered data suggest your GeneXpert Instrument is due for calibration. Request an XpertCheck module calibration kit and perform calibration as instructed. Details on how to perform GeneXpert Instrument calibration can be found at <a href="http://www.stoptb.org/wg/gli/TrainingPackage_Xpert_MTB_RIF.asp">http://www.stoptb.org/wg/gli/TrainingPackage_Xpert_MTB_RIF.asp</a> module 10 MAINTENANCE.');
+                                }
+                                if (!isset($result['qc_done_on_time']) || !$result['qc_done_on_time']) {
+                                    array_push($allComments, 'Entered data suggest monthly maintenance needs to be performed.  Proper and timely monthly maintenance helps to ensure the longevity and accuracy of the GeneXpert Instrument. Details on how to perform GeneXpert Instrument monthly maintenance can be found at <a href="http://www.stoptb.org/wg/gli/TrainingPackage_Xpert_MTB_RIF.asp">http://www.stoptb.org/wg/gli/TrainingPackage_Xpert_MTB_RIF.asp</a> module 10 MAINTENANCE.');
+                                }
+                                if (!isset($result['supervisor_approval']) || $result['supervisor_approval'] == '' || $result['supervisor_approval'] == 'no') {
+                                    array_push($allComments, 'Entered data suggest submitted results were not cross checked by a supervisor or designee.  Supervisory report review is an important step in quality assurance of laboratory testing results and must be incorporated into the workflow of all laboratories to ensure accuracy of all reported results.');
+                                }
```

Comment:
If you're going to extract getting reflexive comments out of the original function, you should use it in the same function that you extracted it from. Otherwise, when the customer ever wants to change the wording on any of these comments or add a new comment, they will need to remember to repeat the code in to separate places.

It looks like this function isn't being called at all though.

Reply:
This was used in https://github.com/SystemOneId/ept/blob/c2d23a654f6d775079ede20ad7ef20da13400467/application/modules/reports/views/scripts/distribution/generate-reports.ajax.phtml#L348-L365 and the main logic is its own position but I used it to get the reflexive comment. Because I can't call it from the .phtml

Reply:
The ticket states that there should be a column for each of the reflexive comments that were generated for the panel:
> A column for each reflexive (automated) comment with an indicator of whether that comment was shown on the participant's report
The report has a generic Reflexive Comment column with all of the comments generated inside that column.

The reason the customer wants each comment separated into it's own column is because they need the ability to sort and filter the worksheet based on a particular reflexive comment.